src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -649,6 +649,23 @@ public function dumpFile($filename, $content)
     }
 
     /**
+     * Atomically appends content to an existing file.
+     *
+     * @param string $filename The file for which to append content
+     * @param string $content  The content to append
+     *
+     * @throws IOException If the file is not writable
+     */
+    public function appendToFile($filename, $content)
+    {
+        if (false !== $originalContent = @file_get_contents($filename)) {
```

Comment:
> which doesn't look like an atomic append to me

The write is atomic, using `file_put_contents` for the read operation would not make it more atomic to me, do I miss something?

> You could refactor a bit the logic inside dumpFile in order to extract it in a private method adding the flag or not.

It was my first intention, but would you avoid creating a tmpfile for append? Otherwise we still have to use `file_get_contents` or doing two `file_put_contents`,